x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/FullTextFunction.java

```diff
@@ -218,8 +225,14 @@ private static void checkFullTextQueryFunctions(LogicalPlan plan, Failures failu
                 m -> "[" + m.functionName() + "] " + m.functionType(),
                 failures
             );
-            checkFullTextSearchDisjunctions(condition, ftf -> "[" + ftf.functionName() + "] " + ftf.functionType(), failures);
             checkFullTextFunctionsParents(condition, failures);
+
+            boolean usesScore = plan.output()
+                .stream()
+                .anyMatch(attr -> attr instanceof MetadataAttribute ma && ma.name().equals(MetadataAttribute.SCORE));
```

Comment:
not sure about this check - it might be safer not to check the plan output, but check if the score metadata field has enabled at any point.

I haven't checked this, but I assume this query will pass the verification:

```
from my-index metadata _score
| where match(a, "foo")
| eval x = _score + 1
| DROP _score
| where match(b, "bar") or length(c) > 100
```
I think `x` here will be the score of `where match(a, "foo")` + 2 - and it still up for debate if that's what we want.
so I think until we get the scoring to work with disjunctions we should just check if the metadata score field was enabled in the first place - it's safer and it's also a temporary limitation.

Reply:
Ah, I see. So you think we should use `plan.inputSet()` instead? I'm unfamiliar with the way attributes are handled, will do some testing with your example query here.

Reply:
okay - let's keep it as it is!